jax/_src/dispatch.py

```diff
@@ -192,7 +192,10 @@ def _device_from_arg_devices(devices: Sequence[Optional[Device]]) -> Optional[De
 def _xla_call_impl(fun: lu.WrappedFun, *args, device, backend, name,
                    donated_invars, inline, keep_unused: bool):
   del inline  # Only used at tracing time
-  arg_specs = unsafe_map(arg_spec, args)
+  if fun.in_type is None:
+    arg_specs = unsafe_map(arg_spec, args)
```

Comment:
Why unsafe_map?

Reply:
I was just keeping it how it was, but I think it's like this for speculative performance reasons. The only place `arg_specs` is consumed is just below.

Reply:
SGTM